Assembly-CSharp/ModLoader.cs

```diff
@@ -110,114 +119,26 @@ public static IEnumerator LoadModsInit(GameObject coroutineHolder)
 
             Logger.APILogger.LogDebug($"Loading assemblies and constructing mods");
 
-            string mods = Path.Combine(managed_path, "Mods");
-
-            string[] files = Directory.GetDirectories(mods)
-                .Except(new string[] { Path.Combine(mods, "Disabled") })
-                .SelectMany(d => Directory.GetFiles(d, "*.dll"))
-                .ToArray();
-
-            Logger.APILogger.LogDebug(string.Join(",\n", files));
-
-            Assembly Resolve(object sender, ResolveEventArgs args)
-            {
-                var asm_name = new AssemblyName(args.Name);
-
-                if (files.FirstOrDefault(x => x.EndsWith($"{asm_name.Name}.dll")) is string path)
-                    return Assembly.LoadFrom(path);
-
-                return null;
-            }
-
-            AppDomain.CurrentDomain.AssemblyResolve += Resolve;
-
-            List<Assembly> asms = new(files.Length);
-
-            // Load all the assemblies first to avoid dependency issues
-            // Dependencies are lazy-loaded, so we won't have attempted loads
-            // until the mod initialization.
-            foreach (string path in files)
-            {
-                Logger.APILogger.LogDebug($"Loading assembly `{path}`");
-
-                try
-                {
-                    asms.Add(Assembly.LoadFrom(path));
-                }
-                catch (FileLoadException e)
-                {
-                    Logger.APILogger.LogError($"Unable to load assembly - {e}");
-                }
-                catch (BadImageFormatException e)
-                {
-                    Logger.APILogger.LogError($"Assembly is bad image. {e}");
-                }
-                catch (PathTooLongException)
-                {
-                    Logger.APILogger.LogError("Unable to load, path to dll is too long!");
-                }
+            string mods = Path.Combine(ManagedPath, "Mods");
+            
+            string[] modDirectories = Directory.GetDirectories(mods)
+                                               .Except([Path.Combine(mods, "Disabled")])
```

Comment:
Guessing we are not using modern enough build tooling for c#13 to be supported here